app/ide-desktop/lib/dashboard/src/hooks/billing/FeaturesConfiguration.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * @file
+ *
+ * Paywall configuration for different plans.
+ */
+
+import type * as text from '#/text'
+
+import * as backend from '#/services/Backend'
+
+/**
+ *
+ */
+export const PAYWALL_FEATURES = {
+  userGroups: 'userGroups',
+  userGroupsFull: 'userGroupsFull',
+} as const
+
+/**
+ * Paywall features.
+ */
+export type PaywallFeatureName = keyof typeof PAYWALL_FEATURES
+
+/**
+ *
+ */
+export type PaywallLevelName = backend.Plan | 'free'
+
+/**
+ *
+ */
+export type PaywallLevelValue =
+  | (0 & { readonly name: PaywallLevelName; readonly label: text.TextId })
+  | (1 & { readonly name: PaywallLevelName; readonly label: text.TextId })
+  | (2 & { readonly name: PaywallLevelName; readonly label: text.TextId })
+  | (3 & { readonly name: PaywallLevelName; readonly label: text.TextId })
+
+export const PAYWALL_LEVELS: Record<PaywallLevelName, PaywallLevelValue> = {
+  free: Object.assign(0, { name: 'free', label: 'freePlanName' } as const),
```

Comment:
Because plans are Inclusive - all features from a level below are included in the levels above. 
So if we want to check that the feature is available, we can simply compare 2 numbers - the user's level and the feature level.

Reply:
we can do that with a property as well. `user.plan.level < plan.level`

Reply:
We can do anything with Javascript. We have infinite possibilities. I see no arguments why `user.plan.level < plan.level` is better than `level < userLevel`

Reply:
it's not! but i'm really, *really* unsure about passing around a `Number` object, just so the code looks a tiny bit cleaner in one specific situation.

Reply:
I my experience checking things like 'level === PaywallLevels.enterprise' or 'level > PaywallLevels.enterpriseEntry' is more than just a `one specific case`. 
Also, since `level` is a primitive type, we don't care if references don't match. 
![CleanShot 2024-05-22 at 19 23 12](https://github.com/enso-org/enso/assets/61194245/cc91bea9-bf6c-433b-91a6-4591e14b6908)

Reply:
I agree this looks __strange__ and unusual, but it definitely has advantages and don't violate agreement we have in javascript